app/code/Magento/WishlistGraphQl/Model/Resolver/WishlistRemoveItemResolver.php

```diff
@@ -0,0 +1,152 @@
+use Magento\Wishlist\Model\ResourceModel\Item\Collection as WishlistItemCollection;
+use Magento\Wishlist\Model\ResourceModel\Item\CollectionFactory as WishlistItemCollectionFactory;
+use Magento\Wishlist\Model\ResourceModel\Wishlist as WishlistResourceModel;
+use Magento\Wishlist\Model\ResourceModel\Item as WishlistItemResourceModel;
+use Magento\Wishlist\Model\Wishlist;
+use Magento\Wishlist\Model\WishlistFactory;
+use Magento\Framework\Exception\LocalizedException;
+
+/**
+ * Remove wishlist items according to the GraphQL schema
+ */
+class WishlistRemoveItemResolver implements ResolverInterface
+{
+    /**
+     * @var WishlistResourceModel
+     */
+    private $wishlistResource;
+
+    /**
+     * @var WishlistItemResourceModel
+     */
+    private $wishlistItemResource;
+
+    /**
+     * @var WishlistFactory
+     */
+    private $wishlistFactory;
+
+    /**
+     * @var WishlistItemCollectionFactory
+     */
+    private $wishlistItemCollectionFactory;
+
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
+
+    /**
+     * @param WishlistResourceModel $wishlistResource
+     * @param WishlistItemResourceModel $wishlistItemResource
+     * @param WishlistFactory $wishlistFactory
+     * @param WishlistItemCollectionFactory $wishlistItemCollectionFactory
+     * @param StoreManagerInterface $storeManager
+     */
+    public function __construct(
+        WishlistResourceModel $wishlistResource,
+        WishlistItemResourceModel $wishlistItemResource,
+        WishlistFactory $wishlistFactory,
+        WishlistItemCollectionFactory $wishlistItemCollectionFactory,
+        StoreManagerInterface $storeManager
+    ) {
+        $this->wishlistResource = $wishlistResource;
+        $this->wishlistItemResource = $wishlistItemResource;
+        $this->wishlistFactory = $wishlistFactory;
+        $this->wishlistItemCollectionFactory = $wishlistItemCollectionFactory;
+        $this->storeManager = $storeManager;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
+        $wishlist = $this->getWishlistById($args['input']['wishlist_id']);
+        $wishlistId = $wishlist->getId();
+
+        $this->deleteWishlistItems($wishlist, $args['input']['wishlist_items_ids']);
+
+        return [
+            'wishlist' => [
+                'id' => $wishlistId,
+                'sharing_code' => $wishlist->getSharingCode(),
+                'updated_at' => $wishlist->getUpdatedAt(),
+                'items_count' => $wishlist->getItemsCount(),
+                'name' => $wishlist->getName(),
+                'model' => $wishlist,
+            ]
+        ];
+    }
+
+    /**
+     * Loads Wishlist by the wishlist id
+     *
+     * @param int $wishlistId
+     * @return Wishlist
+     */
+    private function getWishlistById($wishlistId): Wishlist
```

Comment:
This code is also used in `\Magento\WishlistGraphQl\Model\Resolver\CustomerWishlistResolver::resolve`. We need to move to the separate class in the `execute` method and reuse this code. Example: `\Magento\CustomerGraphQl\Model\Customer\SaveCustomer::execute`

Reply:
Hey @lenaorobei it is just missing this now.

I didn't understand why we should create a `Customer\SaveCustomer` to add the `getWishlistById` method. They are not related, actually.

Can you explain I little bit more about it? I was thinking to add it to a new class in the `WishlistGraphQl\Model` folder, but naming it as something with Wishlist in the name.

Reply:
I meant `Magento\CustomerGraphQl\Model\Customer\SaveCustomer::execute` as an example. In our case it should be `GetWishlistById` class and `execute` method.

Reply:
Hmm, I see. I'll do that.